Kernel/Storage/SD/Commands.h

```diff
@@ -0,0 +1,156 @@
+/*
+ * Copyright (c) 2023, the SerenityOS developers.
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#pragma once
+
+#include <AK/Types.h>
+
+namespace Kernel::SD {
+
+// Commands, defined in PLSS 4.7.4 with the format described in BCM2835 "CMDTM
+// Register"
+enum class CommandIndex {
+    GoIdleState = 0,
+    AllSendCid = 2,
+    SendRelativeAddr = 3,
+    AppSetBusWidth = 6,
+    SelectCard = 7,
+    SendIfCond = 8,
+    SendCsd = 9,
+    SetBlockLen = 16,
+    ReadSingleBlock = 17,
+    ReadMultipleBlock = 18,
+    WriteSingleBlock = 24,
+    WriteMultipleBlock = 25,
+    AppSendOpCond = 41,
+    AppSendScr = 51,
+    AppCmd = 55,
+};
+
+enum class CommandType { Normal,
+    Suspend,
+    Resume,
+    Abort };
+
+enum class ResponseType {
+    NoResponse,
+    ResponseOf136Bits,
+    ResponseOf48Bits,
+    ResponseOf48BitsWithBusy
+};
+struct Command {
+    u8 resp_a : 1;
+    u8 block_count : 1;
+    u8 auto_command : 2;
+    u8 direction : 1;
+    u8 multiblock : 1;
+    u16 resp_b : 10;
+    u8 response_type : 2;
+    u8 res0 : 1;
+    u8 crc_enable : 1;
+    u8 idx_enable : 1;
+    u8 is_data : 1;
+    u8 type : 2;
+    u8 index : 6;
```

Comment:
```suggestion
    ResponseType response_type : 2;
    u8 res0 : 1;
    u8 crc_enable : 1;
    u8 idx_enable : 1;
    u8 is_data : 1;
    u8 type : 2;
    CommandIndex index : 6;
    ```
    This makes the successors way simpler
    (which could also have been a static_cast, which might be checked by UB-san)

Reply:
Neat, didn't know we could do that